### ISSUE TITLE
Plotly Express lines on maps

### ISSUE THREAD

**Issue by nicolaskruchten** [plotly/plotly.py#2601]
Today, `px.line_mapbox` and `px.line_geo` expect rows which are points, which are then split into groups and connected by lines.

In most GIS formats, however, as wrapped by GeoPandas, the entire line feature is often stored in a single row, just like choropleth geometries. We should have either a new pair of PX functions for this (`px.line_feature_mapbox`?) or an option to the existing functions (`px.line_mapbox(geojson=<geojson>)`) to support this option out of the box.

**Comment by nicolaskruchten** [comment]
See also https://github.com/plotly/plotly_express/issues/29

**Comment by harisbal** [comment]
This will be a game changing enhancement. IMHO introducing it as a parameter in `px.line_mapbox` is more preferable.

**Comment by nicolaskruchten** [comment]
Thanks for the enthusiasm @harisbal! What would you see the function signature looking like in this case? I hesitate a bit about making `px.line_mapbox` have in effect different "modes" but maybe this is the right way forward.

**Comment by nicolaskruchten** [comment]
> People working with geospatial data are very used to be able to control the width, apart form the color of the line.

Yes, this is indeed a limitation we have right now... we don't provide fine-grained control over width and color per segment, but per line we could.

**Comment by nicolaskruchten** [comment]
@alexcjohnson what's your instinct on something like this? a new PX function or a new `geojson` argument that in this case wouldn't map straight down to a trace-level `geojson` schema attribute? this new argument would be parsed similarly to the iterative approach in the docs here: https://plotly.com/python/lines-on-mapbox/#lines-on-mapbox-maps-from-geopandas

**Comment by alexcjohnson** [comment]
@nicolaskruchten Feels like various other of the args to `px.line_mapbox` wouldn't make sense with this kind of input data, and if that's the case then perhaps a new function is a cleaner solution.

**Comment by nicolaskruchten** [comment]
Actually all the other args except lat/Lon/line_group would work! Color, line_width, eventually facet etc

**Comment by alexcjohnson** [comment]
Oh OK, I thought it seemed like the data structure would be fairly different so wouldn't support other args the same way - but if it does, then great, just overload the same function!

**Comment by harisbal** [comment]
> Thanks for the enthusiasm @harisbal! What would you see the function signature looking like in this case? I hesitate a bit about making `px.line_mapbox` have in effect different "modes" but maybe this is the right way forward.

I believe that a `mode` argument would be very useful here. The two possible modes would be either `points` or `linestrings`.
In addition, I feel that for the `points` case a `line_order` argument should be also introduced to specify the order of the points.

- px.line_mapbox(df, mode='points', lat='lat, lon='lon', line_group='country', line_order='order') 
- px.line_mapbox(df, mode='linestrings', linestrings=df.geometry)

I'd be glad to help if required. Cheers

**Comment by EsbernJakobsen** [comment]
Just wanted to give this a bump!

Out-of-the-box support for shapely/GeoPandas `LineStrings` would be fantastic. I'm aware you've added some support for Shapely `Points` and `Polygons`. The GeoPandas support example given in [Lines on Maps](https://plotly.com/python/lines-on-maps/) is useful but it doesn't really exploit the strengths of pandas, as it involves unpacking the dataframe using a loop.

Thanks for all the work so far though! 🙂

**Comment by ibhalin** [comment]
I'm bumping too. It would be fantastic !

**Comment by beajifu** [comment]
I want to ask if there is any progress for this feature, because it sounds really good to have support for LineStrings to easily plot a GeoPandas Dataframe. I would appreciate it.

**Comment by gvwilson** [comment]
@archmoj will your recent work address this?
